bpf/Makefile

```diff
@@ -175,7 +175,8 @@ LXC_OPTIONS = \
 	 -DENABLE_HOST_REDIRECT:-DENABLE_IPV4:-DENABLE_IPV6 \
 	 -DENABLE_HOST_REDIRECT:-DENABLE_IPV4:-DENABLE_IPV6:-DENABLE_NAT46 \
 	 -DENABLE_NODEPORT:-DENABLE_IPV4:-DENABLE_IPV6 \
-	 -DENABLE_NODEPORT:-DENABLE_DSR:-DENABLE_IPV4:-DENABLE_IPV6
+	 -DENABLE_NODEPORT:-DENABLE_DSR:-DENABLE_IPV4:-DENABLE_IPV6 \
+	 -DENABLE_IPV4:-DENABLE_IPV4:-DUSE_BPF_PROG_FOR_INGRESS_POLICY
```

Comment:
It's not the first time we miss such a compile test. Could we automate this, even partially? Maybe by grepping for `#ifdefs` macros and checking they are tested in the Makefile?

Reply:
I'd like to abstract it out so that we generate permutations from the same code that configures the datapath on the Golang side. Then we can be a bit more intelligent about the combinations and focus primarily on the sets of options that would actually be configured by the daemon.

But, that's a bit of a bigger / longer task.